astropy/coordinates/sky_coordinate.py

```diff
@@ -900,30 +715,30 @@ def __getattr__(self, attr):
 
     def __setattr__(self, attr, val):
         # This is to make anything available through __getattr__ immutable
-        if "_sky_coord_frame" in self.__dict__:
-            if self._is_name(attr):
-                raise AttributeError(f"'{attr}' is immutable")
+        if attr != "info" and not attr.startswith("_"):
+            if "_sky_coord_frame" in self.__dict__:
+                if self._is_name(attr):
+                    raise AttributeError(f"'{attr}' is immutable")
 
-            if not attr.startswith("_") and hasattr(self._sky_coord_frame, attr):
-                setattr(self._sky_coord_frame, attr, val)
+                if hasattr(self._sky_coord_frame, attr):
+                    setattr(self._sky_coord_frame, attr, val)
+                    return
+
+            if attr in frame_transform_graph.frame_attributes:
+                # All possible frame attributes can be set, but only via a private
+                # variable.  See __getattr__ above.
+                super().__setattr__("_" + attr, val)
+                # Validate it
+                frame_transform_graph.frame_attributes[attr].__get__(self)
+                # And add to set of extra attributes
+                self._extra_frameattr_names |= {attr}
                 return
 
-            frame_cls = frame_transform_graph.lookup_name(attr)
-            if frame_cls is not None and self.frame.is_transformable_to(frame_cls):
```

Comment:
This was mirroring what `__getattr__` does, but surely for `__setattr__` we should just forbid any name that is in the frame transform graph!? (I can change it back, of course).